src/boot/efi/boot.c

```diff
@@ -812,12 +836,30 @@ static BOOLEAN menu_run(
                 case KEYPRESS(0, 0, 'p'):
                 case KEYPRESS(0, 0, 'P'):
                         print_status(config, loaded_image_path);
-                        refresh = TRUE;
+                        clear = TRUE;
                         break;
 
                 case KEYPRESS(EFI_CONTROL_PRESSED, 0, 'l'):
                 case KEYPRESS(EFI_CONTROL_PRESSED, 0, CHAR_CTRL('l')):
-                        refresh = TRUE;
+                        clear = TRUE;
+                        break;
+
+                case KEYPRESS(0, 0, 'r'):
+                        if ((INT64)config->console_mode_efivar < ST->ConOut->Mode->MaxMode &&
+                            ST->ConOut->Mode->Mode == ST->ConOut->Mode->MaxMode - 1) {
+                                config->console_mode_efivar = CONSOLE_MODE_KEEP;
+                                efivar_set(LOADER_GUID, L"LoaderConfigConsoleMode", NULL, EFI_VARIABLE_NON_VOLATILE);
+                                status = StrDuplicate(L"Console mode defined by configuration file.");
+                        } else {
+                                err = console_set_mode(CONSOLE_MODE_NEXT);
+                                config->console_mode_efivar = ST->ConOut->Mode->Mode;
+                                efivar_set_uint_string(LOADER_GUID,
+                                                       L"LoaderConfigConsoleMode",
+                                                       config->console_mode_efivar,
+                                                       EFI_VARIABLE_NON_VOLATILE);
```

Comment:
so, i like that this is now stored in an efi var. but i'd like one tweak here: instead of writing this instantly, write it right before we finish and boot on. why? to minimize nvram write cycles: most likely people will cycle through the available resolutions and once they found a good one, boot on. And I think tha should result in a single nvram write, not many, 

i.e. simply switch console mode, save it in memory, and then when we are done with it sync it to efi var, if found not matching the existing efi var value.